### PR TITLE
Feature/more integration tests

### PR DESCRIPTION
**Pull request by linasi**
- Tests to cover all HcsDid API
- Add tags on test classes to differentiate between unit and integration
- Fix submit message logic

### CODE REVIEW

File: lib/src/main/java/com/hedera/hashgraph/identity/hcs/MessageEnvelope.java [hunk 1]
```diff
@@ -75,6 +75,10 @@ public MessageEnvelope(T message) {
             result.message = null;
         }
 
+        if (jsonNode.has(SIGNATURE_KEY) && jsonNode.get(SIGNATURE_KEY) != null) {
```
Comment (linasi) [inline lib/src/main/java/com/hedera/hashgraph/identity/hcs/MessageEnvelope.java:78]: @vijayshiyani signature was missing when message was loaded for some reason

File: lib/src/main/java/com/hedera/hashgraph/identity/hcs/did/HcsDidTransaction.java [hunk 2]
```diff
@@ -141,7 +141,7 @@ public TransactionId execute(final Client client) throws JsonProcessingException
         new Validator().checkValidationErrors("MessageTransaction execution failed: ", this::validate);
 
         MessageEnvelope<HcsDidMessage> envelope = this.message;
-        byte[] messageContent = envelope.getSignature() == null ? envelope.toJSON().getBytes(StandardCharsets.UTF_8) : envelope.sign(signer);
+        byte[] messageContent = envelope.getSignature() == null ? envelope.sign(signer) : envelope.toJSON().getBytes(StandardCharsets.UTF_8);
```
Comment (linasi) [inline lib/src/main/java/com/hedera/hashgraph/identity/hcs/did/HcsDidTransaction.java:144]: @vijayshiyani this condition was not correct for some reason

File: lib/src/main/java/com/hedera/hashgraph/identity/hcs/MessageEnvelope.java (hunk 1 above)
Comment (vijayshiyani) [reply lib/src/main/java/com/hedera/hashgraph/identity/hcs/MessageEnvelope.java:78]: 👍

**Review by vijayshiyani — APPROVED** [review]
LGTM

**Pull request merged**